File: python/2.7/observer/WeatherData.py
from Subject import *

class WeatherData(Subject):
    def __init__(self):
        super(WeatherData, self).__init__()
        self.observers = []
        self.temperature = 0
        self.humidity = 0
        self.pressure = 0

    def registerObserver(self, o):
        self.observers.append(o)

    def removeObserver(self, o):
        i = observers.index(o)
        if(i >= 0):
            self.observers.remove(i);

    def notifyObservers(self):
        for observer in self.observers:
            observer.update(self.temperature, self.humidity, self.pressure)

    def measurementsChanged(self):
        self.notifyObservers()

    def setMeasurements(self, temperature, humidity, pressure):
        self.temperature = temperature
        self.humidity = humidity
        self.pressure = pressure
        self.measurementsChanged()

    def getTemperature(self):
        return self.temperature

    def getHumidity(self):
        return self.humidity

    def getPressure(self):
        return self.pressure

File: php/5.3/observer/subject/WeatherData.php
<?
include_once("./subject/Subject.php");

class WeatherData implements Subject {
    protected $observers;
    protected $temperature;
    protected $humidity;
    protected $pressure;

    public function __construct() {
        $this->observers = array();
    }

    public function registerObserver(Observer $o) {
        array_push($this->observers, $o);
    }

    public function removeObserver(Observer $o) {
        $i = array_search($o, $this->observers);
        if($i >= 0) {
            unset($this->observers[$i]);
        }
    }

    public function notifyObservers() {
        foreach ($this->observers as $observer)
            $observer->update($this->temperature, $this->humidity, $this->pressure);
    }

    public function measurementsChanged() {
        $this->notifyObservers();
    }

    public function setMeasurements($temperature, $humidity, $pressure) {
        $this->temperature = $temperature;
        $this->humidity = $humidity;
        $this->pressure = $pressure;
        $this->measurementsChanged();
    }

    public function getTemperature() {
        return $this->temperature;
    }

    public function getHumidity() {
        return $this->humidity;
    }

    public function getPressure() {
        return $this->pressure;
    }
}
?>